Extensions themes trigger 'Theme not found: xyz' in log at startup

### Summary

<!-- Please insert a one line summary of the issue below -->
Non-built-in themes (installed using Extensions or just by copying theme.json in `~/.config/zed/themes` are given an error 'Theme not found' in Zed.log.
<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Install theme (Nord in my case) 
2. Set is as default
3. Restart Zed and examine Zed.log

Actual Behavior:

There is 
```
2025-02-10T00:31:10.476533+03:00 [INFO] Opening main db
2025-02-10T00:31:10.514479+03:00 [INFO] Using git binary path: Some("/Applications/Zed Preview.app/Contents/MacOS/git")
2025-02-10T00:31:10.977926+03:00 [ERROR] theme not found: Nordic
2025-02-10T00:31:10.979297+03:00 [INFO] No prompt template overrides directory found at /Users/valerykondakoff/.config/zed/prompt_overrides. Using built-in prompts.
```

while theme is actually loading and exists in the installed extensions list (or theme.json is available in `~/.config/zed/themes` if theme was installed manually). The same happens with other non-build-in themes, so it is not Nord-related.

<img width="1595" alt="Image" src="https://github.com/user-attachments/assets/0ac8179a-646a-4f9b-8727-2ff5ad67d4b9" />


### Zed Version and System Specs

Zed: v0.173.4 (Zed Preview)
OS: macOS 15.3.0
Memory: 32 GiB
Architecture: x86_64

I can reproduce.  On every startup I see the following:
```
2025-02-10T12:45:33.023668-05:00 [ERROR] theme not found: Modus Vivendi Tritanopia
2025-02-10T12:45:33.023745-05:00 [ERROR] icon theme not found: Material Icon Theme
```
I believe what is happening is that Zed is trying to load that theme before the list of themes has actually initialized.  I believe it is harmless (my theme/icon theme loads without issue) but Zed shouldn't output an error on every startup.

Not this only happens when using extension themes / icon themes (e.g. the does not occur with One Dark or other Zed bundled themes.).

> I can reproduce. On every startup I see the following:
> 
> ```
> 2025-02-10T12:45:33.023668-05:00 [ERROR] theme not found: Modus Vivendi Tritanopia
> 2025-02-10T12:45:33.023745-05:00 [ERROR] icon theme not found: Material Icon Theme
> ```
> 
> I believe what is happening is that Zed is trying to load that theme before the list of themes has actually initialized. I believe it is harmless (my theme/icon theme loads without issue) but Zed shouldn't output an error on every startup.
> 
> Not this only happens when using extension themes / icon themes (e.g. the does not occur with One Dark or other Zed bundled themes.).

Yeah, I met the trouble.

```
Zed: v0.174.0 (Zed Dev 2e7a89c5e3300f8e7c245205203dc49188deffad) 
OS: Windows 10.0.26100
Memory: 31.6 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics || Intel Corporation || 101.6127
```

--- a/crates/extension/src/extension_host_proxy.rs
+++ b/crates/extension/src/extension_host_proxy.rs
@@ -96,6 +96,8 @@ impl ExtensionHostProxy {
 }
 
 pub trait ExtensionThemeProxy: Send + Sync + 'static {
+    fn set_extensions_loaded(&self);
+
     fn list_theme_names(&self, theme_path: PathBuf, fs: Arc<dyn Fs>) -> Task<Result<Vec<String>>>;
 
     fn remove_user_themes(&self, themes: Vec<SharedString>);
@@ -123,6 +125,14 @@ pub trait ExtensionThemeProxy: Send + Sync + 'static {
 }
 
 impl ExtensionThemeProxy for ExtensionHostProxy {
+    fn set_extensions_loaded(&self) {
+        let Some(proxy) = self.theme_proxy.read().clone() else {
+            return;
+        };
+
+        proxy.set_extensions_loaded()
+    }
+
     fn list_theme_names(&self, theme_path: PathBuf, fs: Arc<dyn Fs>) -> Task<Result<Vec<String>>> {
         let Some(proxy) = self.theme_proxy.read().clone() else {
             return Task::ready(Ok(Vec::new()));

--- a/crates/extension_host/src/extension_host.rs
+++ b/crates/extension_host/src/extension_host.rs
@@ -1290,6 +1290,7 @@ impl ExtensionStore {
                 }
 
                 this.wasm_extensions.extend(wasm_extensions);
+                this.proxy.set_extensions_loaded();
                 this.proxy.reload_current_theme(cx);
                 this.proxy.reload_current_icon_theme(cx);
             })

--- a/crates/theme/src/registry.rs
+++ b/crates/theme/src/registry.rs
@@ -50,6 +50,8 @@ impl Global for GlobalThemeRegistry {}
 struct ThemeRegistryState {
     themes: HashMap<SharedString, Arc<Theme>>,
     icon_themes: HashMap<SharedString, Arc<IconTheme>>,
+    /// Whether the extensions have been loaded yet.
+    extensions_loaded: bool,
 }
 
 /// The registry for themes.
@@ -82,6 +84,7 @@ impl ThemeRegistry {
             state: RwLock::new(ThemeRegistryState {
                 themes: HashMap::default(),
                 icon_themes: HashMap::default(),
+                extensions_loaded: false,
             }),
             assets,
         };
@@ -98,6 +101,16 @@ impl ThemeRegistry {
             .insert(default_icon_theme.name.clone(), default_icon_theme);
 
         registry
+    }
+
+    /// Returns whether the extensions have been loaded.
+    pub fn extensions_loaded(&self) -> bool {
+        self.state.read().extensions_loaded
+    }
+
+    /// Sets the flag indicating that the extensions have loaded.
+    pub fn set_extensions_loaded(&self) {
+        self.state.write().extensions_loaded = true;
     }
 
     fn insert_theme_families(&self, families: impl IntoIterator<Item = ThemeFamily>) {

--- a/crates/theme/src/settings.rs
+++ b/crates/theme/src/settings.rs
@@ -157,7 +157,11 @@ impl ThemeSettings {
             // If the selected theme doesn't exist, fall back to a default theme
             // based on the system appearance.
             let theme_registry = ThemeRegistry::global(cx);
-            if theme_registry.get(theme_name).ok().is_none() {
+            if let Err(err @ ThemeNotFoundError(_)) = theme_registry.get(theme_name) {
+                if theme_registry.extensions_loaded() {
+                    log::error!("{err}");
+                }
+
                 theme_name = Self::default_theme(*system_appearance);
             };
 
@@ -180,11 +184,13 @@ impl ThemeSettings {
 
             // If the selected icon theme doesn't exist, fall back to the default theme.
             let theme_registry = ThemeRegistry::global(cx);
-            if theme_registry
-                .get_icon_theme(icon_theme_name)
-                .ok()
-                .is_none()
+            if let Err(err @ IconThemeNotFoundError(_)) =
+                theme_registry.get_icon_theme(icon_theme_name)
             {
+                if theme_registry.extensions_loaded() {
+                    log::error!("{err}");
+                }
+
                 icon_theme_name = DEFAULT_ICON_THEME_NAME;
             };
 
@@ -848,7 +854,9 @@ impl settings::Settings for ThemeSettings {
                         this.active_theme = theme;
                     }
                     Err(err @ ThemeNotFoundError(_)) => {
-                        log::error!("{err}");
+                        if themes.extensions_loaded() {
+                            log::error!("{err}");
+                        }
                     }
                 }
             }
@@ -866,7 +874,9 @@ impl settings::Settings for ThemeSettings {
                         this.active_icon_theme = icon_theme;
                     }
                     Err(err @ IconThemeNotFoundError(_)) => {
-                        log::error!("{err}");
+                        if themes.extensions_loaded() {
+                            log::error!("{err}");
+                        }
                     }
                 }
             }

--- a/crates/theme_extension/src/theme_extension.rs
+++ b/crates/theme_extension/src/theme_extension.rs
@@ -24,6 +24,10 @@ struct ThemeRegistryProxy {
 }
 
 impl ExtensionThemeProxy for ThemeRegistryProxy {
+    fn set_extensions_loaded(&self) {
+        self.theme_registry.set_extensions_loaded();
+    }
+
     fn list_theme_names(&self, theme_path: PathBuf, fs: Arc<dyn Fs>) -> Task<Result<Vec<String>>> {
         self.executor.spawn(async move {
             let themes = theme::read_user_theme(&theme_path, fs).await?;

